config/hotreloadable.go

```diff
@@ -1,14 +1,56 @@
 package config
 
-import "time"
+import (
+	"sync"
+	"time"
+)
 
 // RegisterIntConfigVariable registers int config variable
+// Deprecated: use RegisterIntVar or RegisterAtomicIntVar instead
 func RegisterIntConfigVariable(defaultValue int, ptr *int, isHotReloadable bool, valueScale int, keys ...string) {
 	Default.RegisterIntConfigVariable(defaultValue, ptr, isHotReloadable, valueScale, keys...)
 }
 
+// RegisterIntVar registers a not hot-reloadable int config variable
+func RegisterIntVar(defaultValue int, ptr *int, valueScale int, keys ...string) {
+	Default.RegisterIntVar(defaultValue, ptr, valueScale, keys...)
+}
+
+// RegisterAtomicIntVar registers a hot-reloadable int config variable
+func RegisterAtomicIntVar(defaultValue, valueScale int, keys ...string) *Atomic[int] {
```

Comment:
Instead of `Atomic`, shouldn't we be using the term `Reloadable`?

e.g. 
```go
func RegisterReloadableIntVar(defaultValue, valueScale int, keys ...string) *Reloadable[int] {
```

Reloadable should be atomic, but atomic should not necessary be reloadable :)

Reply:
Good point, also atomic is an implementation detail the user doesn't care about.